src/MagnumPlugins/KtxImporter/KtxImporter.cpp

```diff
@@ -0,0 +1,553 @@
+        return;
+    }
+
+    Implementation::KtxHeader header = *reinterpret_cast<const Implementation::KtxHeader*>(data.data());
+    std::size_t offset = sizeof(Implementation::KtxHeader);
+
+    /* Members after supercompressionScheme need endian swapping as well,
+       but we don't need them for now. */
+    Utility::Endianness::littleEndianInPlace(
+        header.vkFormat, header.typeSize,
+        header.pixelWidth, header.pixelHeight, header.pixelDepth,
+        header.layerCount, header.faceCount, header.levelCount,
+        header.supercompressionScheme);
+
+    /* Check magic string to verify this is a KTX file */
+    if(std::memcmp(&header.identifier, Implementation::KtxFileIdentifier, sizeof(header.identifier)) != 0) {
+        Error() << "Trade::KtxImporter::openData(): wrong file signature";
+        return;
+    }
+
+    Containers::Pointer<File> f{new File};
+    f->in = Containers::Array<char>{NoInit, data.size()};
+    Utility::copy(data, f->in);
+
+    /** @todo Support Basis compression */
+    const PixelFormat format = pixelFormat(Vk::PixelFormat(header.vkFormat));
+    if(format == PixelFormat{}) {
+        const CompressedPixelFormat compressedFormat = compressedPixelFormat(Vk::PixelFormat(header.vkFormat));
+        if(compressedFormat == CompressedPixelFormat{}) {
+            Error{} << "Trade::KtxImporter::openData(): unsupported format" << header.vkFormat;
+            return;
+        }
+        f->pixelFormat.compressed = compressedFormat;
+        f->compressed = true;
+    } else {
+        f->pixelFormat.uncompressed = format;
+        f->compressed = false;
+    }
+
+    /** @todo Determine if format needs swizzling */
+    f->needsSwizzle = false;
+
+    /* typeSize is the size of the format's underlying type, not the texel
+       size, e.g. 2 for RG16F. For any sane format it should be a
+       power-of-two between 1 and 8. */
+    if(header.typeSize < 1 || header.typeSize > 8 ||
+        (header.typeSize & (header.typeSize - 1)))
+    {
+        Error{} << "Trade::KtxImporter::openData(): unsupported type size" << header.typeSize;
+        return;
+    }
+    /* Block-compressed formats must have typeSize set to 1 */
+    CORRADE_INTERNAL_ASSERT(!f->compressed || header.typeSize == 1);
+    f->typeSize = header.typeSize;
+
+    /** @todo Support supercompression */
+    if(header.supercompressionScheme != 0) {
+        Error{} << "Trade::KtxImporter::openData(): supercompression is not supported";
+        return;
+    }
+
+    const Vector3ui originalSize{header.pixelWidth, header.pixelHeight, header.pixelDepth};
+    CORRADE_INTERNAL_ASSERT(originalSize.x() > 0);
+
+    if(originalSize.z() > 0) {
+        CORRADE_INTERNAL_ASSERT(originalSize.y() > 0);
+        f->dimensions = 3;
+    } else if(originalSize.y() > 0)
+        f->dimensions = 2;
+    else
+        f->dimensions = 1;
+
+    /** @todo Assert that 3D images can't have depth format */
```

Comment:
I thought it would be handy to have a function that tells you if a `PixelFormat` is a depth and/or stencil format. Might have to do this manually in this plugin to support all Vulkan formats, but maybe an idea for your ever-growing todo list 👀

Reply:
Yes, this makes sense to have directly in Magnum, `isDepthStencilPixelFormat()` for example. I have many such utils for `VertexFormat` already, so a similar set could be for pixel formats.